### PR TITLE
Add StackChanger to Styleguidist

### PR DESCRIPTION
**Pull request by nickgros**
image

### CODE REVIEW

File: src/lib/containers/StackChanger.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import { EndpointObject } from '../utils/functions/getEndpoint'
+import { getStyleguideStack } from '../utils/SynapseClient'
+import { STYLEGUIDE_STACK_COOKIE } from '../utils/SynapseConstants'
+import { RadioGroup } from './widgets/RadioGroup'
+
+type Stack = 'production' | 'staging' | 'development'
+const STACK_MAP: Record<Stack, EndpointObject> = {
+  production: {
+    REPO: 'https://repo-prod.prod.sagebase.org',
+    PORTAL: 'https://synapse.org/',
```
Comment (jay-hodgson) [inline src/lib/containers/StackChanger.tsx:14]: May as well point to www.synapse.org to avoid the redirector

Comment (nickgros) [reply src/lib/containers/StackChanger.tsx:14]:
```suggestion
    PORTAL: 'https://www.synapse.org/',
```

File: src/lib/containers/StackChanger.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    PORTAL: 'https://portal-dev.dev.sagebase.org/',
+  },
+}
+const cookies = new UniversalCookies()
+
+const StackChanger: React.FC = () => {
+  const [stack, setStack] = useState<Stack>('production')
+
+  useEffect(() => {
+    console.log(getStyleguideStack())
```
Comment (jay-hodgson) [inline src/lib/containers/StackChanger.tsx:31]: Intentionally left in?

Comment (nickgros) [reply src/lib/containers/StackChanger.tsx:31]: Nope!
```suggestion
```

File: src/lib/containers/StackChanger.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      OVERRIDE_ENDPOINT_CONFIG: endpointConfig,
+    }
+  }, [])
+
+  useEffect(() => {
+    createStyleguideStackCookie(stack)
+  }, [createStyleguideStackCookie, stack])
+
+  return (
+    <span className={'experimental-mode-wrapper'}>
```
Comment (nickgros) [inline src/lib/containers/StackChanger.tsx:56]:
```suggestion
    <span className={'stack-update-wrapper'}>
```

File: src/lib/containers/StackChanger.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+          },
+          {
+            label: 'Development (portal-dev.dev.sagebase.org)',
+            value: 'development',
+          },
+        ]}
+        onChange={(value: string) => setStack(value as Stack)}
+      />
+    </span>
+  )
```
Comment (nickgros) [inline src/lib/containers/StackChanger.tsx:77]:
```suggestion
    <RadioGroup
      id={`stack-changer-radio-group`}
      value={stack}
      options={[
        {
          label: 'Production (synapse.org)',
          value: 'production',
        },
        {
          label: 'Staging (staging.synapse.org)',
          value: 'staging',
        },
        {
          label: 'Development (portal-dev.dev.sagebase.org)',
          value: 'development',
        },
      ]}
      onChange={(value: string) => setStack(value as Stack)}
    />
  )
```

File: src/lib/containers/StackChanger.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
+import { isEqual } from 'lodash-es'
+import * as React from 'react'
+import { useEffect, useState } from 'react'
+import UniversalCookies from 'universal-cookie'
```
Comment (jay-hodgson) [inline src/lib/containers/StackChanger.tsx:4]: Reason for choosing a cookie persistence store vs local storage?

Comment (nickgros) [reply src/lib/containers/StackChanger.tsx:4]: No reason, just didn't think about it! localStorage would be better

File: styleguide.setup.js [hunk 6]
```diff
@@ -36,6 +37,15 @@ import {
 import React from 'react'
 import ReactDOM from 'react-dom'
 
+const stackCookie = JSON.parse(
```
Comment (jay-hodgson) [inline styleguide.setup.js:40]: (constant name could be improved after change to localstorage)

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**